CONTRIBUTING.md

```diff
@@ -16,11 +16,10 @@ pip install -e .[test]
 jupyter serverextension enable --sys-prefix jupyter_server_proxy
 ```
 
-Before running tests, you need a [notebook
-server](https://github.com/jupyter/notebook) that we can test against.
+Before running tests, you need a server that we can test against.
 
 ```
-JUPYTER_TOKEN=secret jupyter-notebook --config=./tests/resources/jupyter_server_config.py --no-browser
+JUPYTER_TOKEN=secret jupyter-lab --config=./tests/resources/jupyter_server_config.py --no-browser
```

Comment:
Instead of this rather convoluted invocation  (which wouldn't work on e.g. windows): we could hoist the server to a (parametrized) fixture which started either the `notebook` or `lab` process, depending on availability, and by default tested both.

Reply:
That could be quite convenient. Maybe then it would be possible to coordinate which `notebook`/`lab` port to use. (Often I would try to run tests and port 8888 would be already in use.)

Reply:
haha, yeah, ports. On `robotframework-jupyterlibrary`, we actually [randomize the port](https://github.com/robots-from-jupyter/robotframework-jupyterlibrary/blob/master/src/JupyterLibrary/keywords/server.py#L245) and and the token every time.

Another thing, _particularly_ to which effective use of `jupyter-server-proxy` is _particularly_ sensitive, is `base_url`. It might be worth adding a "nasty" one to the matrix.

Once "parametrize all the tests" goes a bit too far, though, `hypothesis` can be super handy... but one typically wouldn't want to restart process (much less the browser) if it's going to run a couple thousand times. But i feel like `jupyter-(lab|notebook)` and `--base-url="(/|/a/very long/prefix/💩/)"` may well be warranted.

Reply:
This sounds like a good idea. Would it make sense to open a separate issue for it? (Unfortunately I don't have any time to work on such stuff at the moment.)

Reply:
issue up! #309